### PR TITLE
[recipe] feat: add yarn + long context recipe

### PR DESCRIPTION
**Pull request by RayenTian**
### Summary

- Add a new `qwen3_600m_sft_128k_config()` recipe that extends the base Qwen3-600M SFT config with 128K sequence length and context parallelism (CP=8).
- Add a ready-to-run example script (`examples/long_context/finetune_qwen3_600m_128k.py`) for launching 128K SFT training with CLI + Hydra-style overrides.

### Changes

**New recipe: `qwen3_600m_sft_128k_config()`** (`src/megatron/bridge/recipes/qwen/qwen3.py`)

- Sequence length set to 131072 (128K) with packed sequences.
- Parallelism: TP=1, CP=8 (minimum 8 GPUs), SP disabled.
- Uses all-to-all (Ulysses) CP comm (`cp_comm_type = "a2a"`) to avoid NaN gradients in backward pass.
- `calculate_per_token_loss = True` to handle CP > 1 in SFT where some CP ranks may have all tokens masked.
- `average_in_collective = False` (required when `calculate_per_token_loss=True`).
- Low constant learning rate (`1e-6`) suitable for long-context SFT fine-tuning.
- Distributed optimizer and precision-aware optimizer disabled; gradients and params kept in FP32.

**New example script** (`examples/long_context/finetune_qwen3_600m_128k.py`)

- End-to-end entry point supporting `--config-file` (YAML overrides) and Hydra-style dot-notation CLI overrides.
- Supports mock data (no args) or custom data via overrides.

**Export** (`src/megatron/bridge/recipes/qwen/__init__.py`)

- Exports `qwen3_600m_sft_128k_config` from the `qwen` recipes package.

### Results
image

# Before your PR is "Ready for review"

**Pre checks**:

- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA-NeMo/Megatron-Bridge/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?

If you haven't finished some of the above items you can still open "Draft" PR.

# Additional Information

- Related to # (issue)

## Summary by CodeRabbit

* **New Features**
  * Added a new executable training script for fine-tuning Qwen3 600M models with extended 128K sequence length support.
  * Introduced configuration preset for Qwen3 600M SFT training optimized for long-context scenarios.
  * Training script supports YAML configuration files and command-line parameter overrides for flexible customization.

### CODE REVIEW

File: src/megatron/bridge/recipes/qwen/qwen3.py [hunk 1]
```diff
@@ -587,6 +587,113 @@ def qwen3_600m_sft_config() -> ConfigContainer:
     return cfg
 
 
+def qwen3_600m_sft_yarn_128k_config() -> ConfigContainer:
```
Comment (yaoyu-33) [inline src/megatron/bridge/recipes/qwen/qwen3.py]: remove yarn from naming, just 128k

Comment (RayenTian) [reply src/megatron/bridge/recipes/qwen/qwen3.py]: Done.

**Comment by RayenTian** [comment]
/ok to test [903b5fc](https://github.com/NVIDIA-NeMo/Megatron-Bridge/pull/2951/commits/903b5fcfa3e418ce24c44d57c5bb02661f46eb6b)

File: examples/long_context/finetune_qwen3_600m_128k.py [hunk 2]
```diff
@@ -0,0 +1,120 @@
+#!/usr/bin/env python3
```
Comment (yaoyu-33) [inline examples/long_context/finetune_qwen3_600m_128k.py]: similar to model gpt-oss example, let's provide a "sft.sh" using scripts/training/run_recipe.

**Review by cuichenx — APPROVED** [review]
